src/App.jsx

```diff
@@ -14,57 +16,77 @@ export const goods = [
+export const App = () => {
+  const [val, setGood] = useState(goods[8]);
+  const [title, setTitle] = useState(`${val} is selected`);
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {`${title}`}
+        {val ? (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setGood('');
+              setTitle('No goods selected');
+            }}
+          />
+        ) : (
+          ''
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={classNames({
+                'has-background-success-light': val === good,
+              })}
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+              <td>
+                {val === good ? (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className={classNames('button', {
+                      'is-info': val === good,
+                    })}
+                    onClick={() => {
+                      setGood('');
+                      setTitle('No goods selected');
+                    }}
+                  >
+                    -
+                  </button>
+                ) : (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={classNames('button', {
+                      'is-info': val === good,
+                    })}
+                    onClick={() => {
+                      setGood(good);
+                      setTitle(`${good} is selected`);
+                    }}
+                  >
+                    +
+                  </button>
```

Comment:
This component can be merged into one where you will conditionally set good or change name of button